from fastapi import FastAPI, UploadFile, File, HTTPException
from fastapi.responses import JSONResponse
from PIL import Image
import io
from model import predict

app = FastAPI(title="üêü Fish Species Classifier API")

@app.get("/")
def root():
    return {"message": "Bienvenue sur l‚ÄôAPI de classification de poissons üê†"}

@app.post("/predict")
async def classify(file: UploadFile = File(...)):
    try:
        # Lecture de l‚Äôimage envoy√©e
        image_bytes = await file.read()
        image = Image.open(io.BytesIO(image_bytes)).convert("RGB")

        # Pr√©diction
        label, confidence = predict(image)

        return JSONResponse({
            "prediction": label,
            "confidence": round(confidence * 100, 2)
        })

    except Exception as e:
        raise HTTPException(status_code=500, detail=f"Erreur lors de la pr√©diction : {str(e)}")
